### PR TITLE
Fixes #79 - Don't overwrite EFI grub.cfg

### LINKED ISSUE #79: Don't overwrite EFI grub.cfg
This module uses an exec to `grub-mkconfig -o /boot/efi/EFI/ubuntu/grub.cfg` which overrides the default contents, but this file should not contain a "normal" grub configuration, instead the EFI grub.cfg should look like this to reference `/boot/grub/grub.cfg` on the right partition:

```
search.fs_uuid {uuid} root hd0,{partition}
set prefix=($root)'/boot/grub'
configfile $prefix/grub.cfg
```

Because this module overwrites the EFI grub.cfg, and the kernel postinstall scripts only call `grub-update`, which only updates `/boot/grub/grub.cfg`, this module effectively breaks kernel updates as they will never show up in grub.

### PR DESCRIPTION
**Pull request by Jayfrown**
This module overwrites the EFI grub.cfg by calling
`grub-mkconfig -o /boot/efi/EFI/ubuntu/grub.cfg` whenever the
`/etc/default/grub` is altered in a puppet run, but this file should not
contain a "normal" grub.cfg configuration.

Instead, the EFI grub config should reference the "real" grub.cfg as
follows:

```
search.fs_uuid {uuid} root hd0,{partition}
set prefix=($root)'/boot/grub'
configfile $prefix/grub.cfg
```

This ensures machines booting EFI effectively still use
`/boot/grub/grub.cfg` which is the only file that is updated by kernel
postinstall scripts.

In other words, this module currently effectively breaks kernel
updates by preventing them from showing up in grub.

### CODE REVIEW

**Comment by Jayfrown** [comment]
I've only tested this behavior on Ubuntu, but I imagine it works largely the same on other distros, as the EFI config should always support the `configfile` directive.

**Comment by goldyfruit** [comment]
@Jayfrown sorry I missed your PR... I just checked on Red Hat 8, Fedora 36 and Debian 11, your change seems to be OK.

**Review by goldyfruit — APPROVED** [review]

**Comment by goldyfruit** [comment]
New release pushed on Puppet Forge, thanks for the PR.
